### PR TITLE
Enable QUIC client by default. Add arg to disable QUIC client.

### PR DESCRIPTION
**Pull request by willhickey**
#### Problem
* As of v1.10.32 the validator listens for QUIC connections by default
* The validator can forward transactions via quic if run with `--tpu-use-quic`, but doesn't by default

#### Summary of Changes
* Switch the default client behavior to use QUIC
* Add argument to disable the quic client

### CODE REVIEW

**Review by sakridge — APPROVED** [review]
thanks!

File: validator/src/main.rs [hunk 1]
```diff
@@ -1205,10 +1205,10 @@ pub fn main() {
                 .help("Milliseconds to wait in the TPU receiver for packet coalescing."),
         )
         .arg(
```
Comment (mvines) [inline validator/src/main.rs:1207]: Retain the old `tpu-use-quic` arg please, as a `.hidden(true).conflicts_with("tpu_disable_quic")`, for a graceful update for those who are already using the old arg

**Comment by pgarg66** [comment]
Looks like `test_connection_cache` is failing due to a bug in the test. It's failing specifically when QUIC is enabled.

The test creates a list of IPAddresses (and ports), creates connections using the addresses, and then looks up the table using address as the key. With QUIC, we are adding an offset (6) to the port. So the key (address/port) in the table is different than what the test had created.

```
Looking up 215.131.162.105:80
First entry in the table: 215.131.162.105:86
```
Maybe the test can be disabled as part of this PR, and an issue can be created to fix the test as a follow up PR?

File: validator/src/main.rs (hunk 1 above)
Comment (willhickey) [reply validator/src/main.rs]: Done. Thanks for the catch

**Comment by behzadnouri** [comment]
@willhickey Seems like this commit is causing these two tests to timeout in coverage CI:
```
banking_stage::tests::test_forwarder_budget
banking_stage::tests::test_handle_forwarding
```
The CI was also failing:
https://buildkite.com/solana-labs/solana/builds/78901

**Comment by pgarg66** [comment]
I briefly looked at these two tests. The receiver for the forwarded transactions is UDP based (in the tests). Since now we are using QUIC for forwarding the transaction, nothing is received on the UDP port, and it waits for ever.

```
                recv_socket
                    .set_nonblocking(expected_num_forwarded == 0)
                    .unwrap();

                let mut packets = vec![Packet::default(); 2];
                let num_received = recv_mmsg(recv_socket, &mut packets[..]).unwrap_or_default();
```

The solution will be start a QUIC streamer on the receiver port, and use the channel provided to the streamer to receive the packets.

**Comment by behzadnouri** [comment]
@willhickey I see that in https://github.com/solana-labs/solana/pull/26912 you are commenting out the tests.

There is no tracking issues to fix those tests and you commenting them out prevents compile time checks that code still compiles.

* Can you please instead of commenting out tests use `#[ignore]` so that the code is at least is compile checked?
* Can you please open a tracking issue to fix the test if you are ignoring tests?

**Comment by behzadnouri** [comment]
This commit also broke:
```
cli::tests::test_cli_parse_dos_valid_signatures
```

**Comment by willhickey** [comment]
> This commit also broke:
> 
> ```
> cli::tests::test_cli_parse_dos_valid_signatures
> ```

Yeah... there are a bunch of broken tests.

PR #26913 has landed to revert this so CI should be unblocked. I'll work on cleaning up the tests and open a new PR

**Pull request merged**